Studio-Lovelies/GG-JointJustice-Unity
# Refactor proposal: Deduct game mode from .ink script

## Summary
To facilitate the idea of "1 scene per episode", Unity needs to be able to deduct the game state and find all required resources **only** by parsing a .ink-file. 
This PR moves one still existing dependency from the Unity editor to the `.ink` file: the `DialogueControllerMode`, which determines if a scene is a cross examination or pure dialogue.

(Also as this was referred to as `ScriptType`, `DialogueMode` and `DialogueControllerMode` in various places across the codebase, I have unified this to `GameMode` [since a change of this variable effectively changes the entire rules of the game] but I am open to other suggestions too.)

Every .ink-script is now required to start with the `&MODE:` action line set to either `&MODE:CrossExamination` or `&MODE:Dialogue`.
This PR updates all .ink-files to `Dialogue`, except for `Ross_Cool_X.ink` which set to `CrossExamination`.
Also automated tests were added, that ensure human-readable error messages are printed if an `.ink`-file does not meet this new requirement.

## Before/after screenshots and/or animated gif
Before: 
![image](https://user-images.githubusercontent.com/1689033/147863253-f48b6b0f-ffea-4be8-a7bb-d1db0e2a79be.png)
After: 
![image](https://user-images.githubusercontent.com/1689033/147863235-c5c9e06c-ebf1-47ad-bbd9-9126fe620fd2.png)


## Testing instructions

1. Run the newly added `Dialogue` tests 
2. Run any previously working scene (I've used `CrossExamination - Test Scene` and `Evidence Menu - Test Scene`)
3. Ensure the correct UI elements are shown and the game still behaves like before this change

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
